src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/InterventionFilterRepositoryImpl.kt

```diff
@@ -61,13 +62,18 @@ class InterventionFilterRepositoryImpl(
     return expression.`in`(npsRegions)
   }
 
-  private fun getGenderPredicate(criteriaBuilder: CriteriaBuilder?, root: Root<Intervention>?, gender: String?): Predicate? {
-
-    if (gender.isNullOrEmpty()) {
+  private fun getAllowsFemalePredicate(criteriaBuilder: CriteriaBuilder, root: Root<Intervention>, allowsFemale: Boolean?): Predicate? {
+    if (allowsFemale == null) {
       return null
     }
+    return criteriaBuilder.equal(root.get<Any>("dynamicFrameworkContract")?.get<Any>("allowsFemale"), allowsFemale)
```

Comment:
do the types matter here? in the other predicate we specify some of the types in the calls to `get`

Reply:
Good spot. I don't think it actually affected anything, but i've updated all of them in this file to be more specific.